tools/c7n_azure/tests_azure/test_policy_mode.py

```diff
@@ -59,15 +61,62 @@ def test_azure_function_event_mode_too_many_events_throws(self):
                         'type': FUNCTION_EVENT_TRIGGER_MODE,
                         'events': [
                             'VmWrite',
-                            'AppServicePlanWrite',
-                            'CognitiveServiceWrite',
+                            {
+                                'resourceProvider': 'Microsoft.Compute/virtualMachines/',
+                                'event': 'delete'
+                            },
+                            {
+                                'resourceProvider': 'Microsoft.Compute/virtualMachines/',
+                                'event': 'powerOff/action'
+                            },
+                            {
+                                'resourceProvider': 'Microsoft.Compute/virtualMachines/',
+                                'event': 'reimage/action'
+                            },
+                            {
+                                'resourceProvider': 'Microsoft.Compute/virtualMachines/',
+                                'event': 'redeploy/action'
+                            },
+                            {
+                                'resourceProvider': 'Microsoft.Compute/virtualMachines/',
+                                'event': 'start/action'
+                            }
+                        ]
+                    }
+                }, validate=True)
+
+    def test_azure_function_event_mode_incorrect_event_type(self):
+        with self.sign_out_patch():
```

Comment:
nit: not sure where it started, but we don't need `self.sign_out_patch()`

Reply:
I'm going to keep it. The sign_out_path() was created because we were creating sessions before runtime (in init and validations) so the schema validation tests were added to prevent that happening. It doesn't harm the tests and adds another possible catch in the validation logic.